functests/2_performance_update/kubelet.go

```diff
@@ -0,0 +1,177 @@
+				evictionMemoryInt, err := strconv.ParseInt(strings.TrimSuffix(evictionMemory, "Mi"), 10, 64)
+				kubeReservedMemoryInt, err := strconv.ParseInt(strings.TrimSuffix(kubeReserved, "Mi"), 10, 64)
+				systemReservedResource := resource.NewQuantity(300*1024*1024, resource.BinarySI)
+				kubeReservedMemoryResource := resource.NewQuantity(kubeReservedMemoryInt*1024*1024, resource.BinarySI)
+				evictionMemoryResource := resource.NewQuantity(evictionMemoryInt*1024*1024, resource.BinarySI)
+				totalKubeMemory := systemReservedResource.MilliValue() + kubeReservedMemoryResource.MilliValue() + evictionMemoryResource.MilliValue()
+				calculatedAllocatable := totalCapactity - totalKubeMemory
+				currentAllocatable := node.Status.Allocatable.Memory().MilliValue()
+				Expect(calculatedAllocatable).To(Equal(currentAllocatable))
+			}
+		})
+		It("[test_id:45495] Test setting PAO managed parameters", func() {
+			paoAnnotation := map[string]string{
+				"kubeletconfig.experimental": "{\"topologyManagerPolicy\":\"single-numa-node\"}",
+			}
+			profile.SetAnnotations(paoAnnotation)
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			profiles.UpdateWithRetry(profile)
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+			By("Waiting when mcp finishes updates")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+			for _, node := range workerRTNodes {
+				kubeletConfig, err := nodes.GetKubeletConfig(&node)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(kubeletConfig.TopologyManagerPolicy).To(Equal("single-numa-node"))
+			}
+		})
+		It("[test_id:45489] Verify settings are reverted to default profile", func() {
+			By("Reverting the Profile")
+			Expect(testclient.Client.Patch(context.TODO(), profile,
+				client.RawPatch(
+					types.JSONPatchType,
+					[]byte(fmt.Sprintf(`[{ "op": "remove", "path": "/metadata/annotations/kubeletconfig.experimental"}]`)),
+				),
+			)).ToNot(HaveOccurred())
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+			kubeletArguments := []string{"/bin/bash", "-c", "ps -ef | grep kubelet | grep config"}
+			for _, node := range workerRTNodes {
+				kubeletConfig, err := nodes.GetKubeletConfig(&node)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(kubeletConfig.AllowedUnsafeSysctls).To(Equal(nil))
+				Expect(kubeletConfig.KubeReserved["memory"]).ToNot(Equal("768Mi"))
+				stdout, err := nodes.ExecCommandOnNode(kubeletArguments, &node)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(strings.Contains(stdout, "300Mi")).ToNot(BeTrue())
+				Expect(kubeletConfig.ImageMinimumGCAge.Seconds()).ToNot(Equal(180))
+			}
+
+		})
+
+	})
+})
+
+func kubeletSliceValues(values []string, str string) bool {
```

Comment:
changing matcher (see above about `gomega.ContainElements`) will make this obsolete

Reply:
In the latest patch i have addressed this